Code/Framework/AzCore/AzCore/Dependency/Dependency.h

```diff
@@ -49,7 +48,7 @@ namespace AzFramework
                 GreaterThan = 1 << 0,
                 LessThan = 1 << 1,
                 EqualTo = 1 << 2,
-                // Special operators
+                // Special operators ~> and ~=
```

Comment:
We can get rid of the TwiddleWakka, that isn't needed for the semantic version support we envision using.

Reply:
I left it in for backwards compatibility with the XMLDependency code - if that is no longer needed then I'm down with removing it.

Reply:
@AMZN-stankowi Do you know if the XMLDependency code supports the `~>` version specifier?

It know it was used by CrySystem to check version dependencies in Lumberyard that used the old gem format, but would need clarification if the XMLDependency builder is using it.

Reply:
It's used by the XML schema system. This is a data driven solution for searching for product dependencies in XML files. This was most valuable with CryEngine code because there were many different XML file types supported by the engine, but it continues to be useful for anyone who has an XML file they don't process to binary and instead just copy to the asset cache and load at runtime.

Two good entry points for that system are here: https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Asset/XmlSchemaAsset.cpp and here: https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/XmlBuilderWorker/XmlBuilderWorker.cpp

We've talked about expanding this to JSON, CSV, and other formats that are often just copied to the cache and not modified in the future. There's no immediate need for that, but I figured I would mention because the system was built to be grown forward to handle other data formats, too.